### PR TITLE
RET-1127 chore: update Buildroot to 2022.04.2

### PR DESCRIPTION
**Pull request by vegano1**
[RET-1127](https://opentrons.atlassian.net/browse/RET-1127)

# Overview
This change updates Buildroot from 2018.11 -> 2022.04.2, which updates the kernel to 5.10 and most of our packages. This means we will need a full regression to capture any potential issues.

# Changelog
### Major Changes:
- Kernel 4.14 -> 5.10
- Python 3.7 -> 3.10.4
- dropbear 2018.2 -> 2020.81
- wpa_supplicant 2.9 -> 2.10
- See full list of packages changes can be found [here](https://git.busybox.net/buildroot/plain/CHANGES?id=2022.02.4)

### Python Packages:
- pip 18.1 -> 21.2.4
- setuptools 44.0.0 -> 60.7.0
- numpy 1.15.1 -> 1.21.2
- Full list of python packages changes can be found [here](https://docs.google.com/spreadsheets/d/1FN59uN--Si92tMGVR71_oMAs_g2sSpIO-TDr2uHa7hw/edit?usp=sharing)

# Review requests
- Make sure there aren't any obvious issues with the changes

# Risk Assessment
High, this will update the kernel to 5.10 and a bunch of packages and python modules, will need full regression test.

### Environments
All below tests should be run on the following environments:
- [ ] Update/Downgrade? using update-server and a build output
- [ ] Do a fresh flash of an sd card

### Tests
- [ ] Connect to an 802.11 network using wpa2-psk
- [ ] Connect to an 802.11 network using RADIUS
- [ ] ssh to the dev
- [ ] retrieve an HTTPS webpage
- [ ] Run a protocol, especially one that connects to a module
- [ ] Test the robot lights
- [ ] Interact with the robot via jupyter
- [ ] Install a python package in /var
- [ ] make sure we can query the RTC for OT2 Refresh with `hwclock` while on the robot
- [ ] make sure you can rename a robot through the Opentrons App

# Notes:
- Removes RTC-rv3028 patches as it is now officially supported in kernel 5.10
- Removed manual config.txt changes as buildroot lets you provide your own
- Does NOT include pi4 support, this will be added in a separate pr as its unrelated to updating buildroot

[RET-1127]: https://opentrons.atlassian.net/browse/RET-1127?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: board/opentrons/ot2/config.txt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+enable_uart=1
+#dtoverlay=miniuart-bt	 # NOT WORKING
+
+# How much memory in MB to assign to the GPU on Pi models having
+# 256, 512 or 1024 MB total memory
+gpu_mem_256=100
+gpu_mem_512=100
+gpu_mem_1024=100
+
+dtoverlay=gpio-revision-bits
```
Comment (sfoster1) [inline board/opentrons/ot2/config.txt:26]: do these assignments append or overwrite? do we need to change this to `dtoverlay=gpio-revision-bits i2c-rtc,rv3028`?

**Review by sfoster1 — APPROVED** [review]
Aside from the question about the devicetree config, I reviewed the defconfig and the board tree and they look good. Beyond that, it's a question of testing I think.

File: board/opentrons/ot2/config.txt (hunk 1 above)
Comment (vegano1) [reply board/opentrons/ot2/config.txt:26]: The assignments append

**Review by SyntaxColoring — COMMENTED** [review]
🥳 

I frankly have no idea how to review this other than to flash it to a robot, play around, and see if anything weird happens. But some thoughts:

* I added robot renaming to the test checklist, since that touches Avahi, which had updates.
* I think after this merges, we can undo this piece of devops tech debt:
https://github.com/Opentrons/buildroot/pull/132 I think that should happen in a separate PR. ~~Do you want it tracked in Jira somewhere?~~ Ticketed as RET-1177.
* Do we know what it will take to get all the `Opentrons/opentrons` stuff to run with these changes? Is it as simple as changing dev and CI setups to Python 3.10 and updating dependencies?
    * @mcous A while ago, I remember we had to disable some tests on Python 3.10 because they were failing for mysterious reasons. Do you remember where those were? Have those been resolved?

**Comment by mcous** [comment]
> A while ago, I remember we had to disable some tests on Python 3.10 because they were failing for mysterious reasons. Do you remember where those were? Have those been resolved?

There are two things I think you might be talking about here:

- A weird Pydantic parsing failure that only happened on Python 3.8
    - Resolved by replacing the JSON v6 command in question, `delay`, with separate `waitForResume` and `waitForDuration` commands
- Various test failures in `shared-data/python` with tests that use the `typegaurd` module
    - Worked around with `xfail`
    - `typegaurd` is only used in the tests, not the source under test itself

**Comment by vegano1** [comment]
@SyntaxColoring 
- I did notice we locked to Debian 10 for the container, we should be able to fix that. Can you make a ticket.
- Im not sure if there are any specific breaking changes because of this python version, however I did not encounter any obvious issues when running opentrons edge on this build in emulation mode.

**Pull request closed without merging**